packages/renderer/src/lib/preferences/docker-compat/PreferencesDockerCompatibilityDockerContext.spec.ts

```diff
@@ -0,0 +1,138 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import '@testing-library/jest-dom/vitest';
+
+import { render, screen } from '@testing-library/svelte';
+import userEvent from '@testing-library/user-event';
+import { tick } from 'svelte';
+import { beforeEach, expect, type Mock, test, vi } from 'vitest';
+
+import type { DockerContextInfo } from '/@api/docker-compatibility-info';
+
+import PreferencesDockerCompatibilityDockerContext from '././PreferencesDockerCompatibilityDockerContext.svelte';
+
+const contextInfos = [
+  {
+    name: 'default',
+    isCurrentContext: false,
+    metadata: { description: 'default context' },
+    endpoints: {
+      docker: {
+        host: 'unix:///var/run/docker.sock',
+      },
+    },
+  },
+  {
+    name: 'podman',
+    isCurrentContext: true,
+    metadata: { description: 'podman context' },
+    endpoints: {
+      docker: {
+        host: 'unix:///var/podman.sock',
+      },
+    },
+  },
+];
+
+const getDockerContextsMock: Mock<() => Promise<DockerContextInfo[]>> = vi.fn();
+const switchDockerContextMock: Mock<(contextName: string) => Promise<void>> = vi.fn();
+
+beforeEach(() => {
+  vi.resetAllMocks();
+
+  Object.defineProperty(global, 'window', {
+    value: {
+      navigator: {
+        clipboard: {
+          writeText: vi.fn(),
+        },
+      },
+      getDockerContexts: getDockerContextsMock,
+      switchDockerContext: switchDockerContextMock,
+    },
+    writable: true,
+  });
+});
+
+test('only default context', async () => {
+  getDockerContextsMock.mockResolvedValue([
+    {
+      name: 'default',
+      isCurrentContext: true,
+      metadata: { description: 'default context' },
+      endpoints: {
+        docker: {
+          host: 'unix:///var/run/docker.sock',
```

Comment:
should these be const's since it's been used multiple times?

Reply:
applied constant